util/image_viewer/spotfinder_frame.py

```diff
@@ -96,8 +96,51 @@ def __init__(self, *args, **kwds):
 
         super().__init__(*args, **kwds)
 
+        self.viewing_stills = True
+        for explst in self.experiments:
+            for exp in explst:
+                if exp.scan is not None or exp.goniometer is not None:
+                    self.viewing_stills = False
+                    break
+
+        if self.viewing_stills:
+            is_multi_shot_exp = [len(El) > 1 for El in self.experiments]
+            if any(is_multi_shot_exp):
+                assert len(self.experiments) == 1
+                if self.reflections:
+                    assert len(self.reflections) == 1
+                    assert len(self.experiments[0]) == len(
+                        set(self.reflections[0]["id"])
+                    )
+                expts = []
+                refls = []
+                for i_expt, expt in enumerate(self.experiments[0]):
+                    print(
+                        "ravel experiments %d / %d"
+                        % (i_expt + 1, len(self.experiments[0]))
+                    )
+                    El = ExperimentList()
+                    El.append(expt)
+                    expts.append(El)
```

Comment:
Any chance you could find different names for `expts`, `expt`, and `El`, which are all very similar and confusing.

eg. `expts` appears to be something like `new_experiments`...